### PR TITLE
feat(cypress):Add the cypress-visual regression check to the centreon CI/CD testing step

### PR DESCRIPTION
**Pull request by Noha-ElAbrouki**
**-  **upgrade cypress to version 10****  

**- integrate cypress ct to ci** 🥳 🥳 

update:
cypress no longer return exit code 0 for failed tests .. not need to customize the test runner behavior .

- **useful links :** 

https://glebbahmutov.com/blog/wrap-cypress-using-npm/
https://glebbahmutov.com/blog/

### CODE REVIEW

File: cypress.config.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    setupNodeEvents: (on, config) => {
+      addMatchImageSnapshotPlugin(on, config);
+    },
+    specPattern: './www/front_src/src/**/*.cypress.spec.tsx',
+    supportFile: './cypress/support/component.tsx',
+  },
+  reporter: 'junit',
+  reporterOptions: {
+    mochaFile: 'cypress/results/cypress-fe.xml',
+    toConsole: true,
```
Comment (Thebarda) [inline cypress.config.ts]:
```suggestion
```

File: cypress.d.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { MountOptions, MountReturn } from 'cypress/react';
+import { MemoryRouterProps } from 'react-router-dom';
+
+declare global {
+  namespace Cypress {
+    interface Chainable {
+      /**
```
Comment (Thebarda) [inline cypress.d.ts]: You can remove this comment

File: cypress/support/component-index.html [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width,initial-scale=1.0">
+    <title>Components App</title>
+
```
Comment (Thebarda) [inline cypress/support/component-index.html]:
```suggestion
```

File: cypress/support/component-index.html [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+<html>
+  <head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width,initial-scale=1.0">
+    <title>Components App</title>
+    
+  </head>
+  <body>
+
```
Comment (Thebarda) [inline cypress/support/component-index.html]:
```suggestion
```

File: package.json [hunk 5]
```diff
@@ -12,8 +12,8 @@
     "test:watch": "npm test -- --watch",
     "build:watch": "cross-env WEBPACK_ENV=serve webpack serve --mode development --config ./webpack.config.dev.js",
     "test:coverage": "jest --coverage",
-    "cypress:cli": "npx cypress run-ct",
-    "cypress:ui": "npx cypress open-ct"
+    "cypress:cli": "node wrapCypress.ts cypress  run --component",
+    "cypress:ui": "npx cypress open --component"
```
Comment (Thebarda) [inline package.json]:
```suggestion
    "cypress:ui": "cypress open --component --browser=chrome"
```

File: package.json [hunk 6]
```diff
@@ -12,8 +12,8 @@
     "test:watch": "npm test -- --watch",
     "build:watch": "cross-env WEBPACK_ENV=serve webpack serve --mode development --config ./webpack.config.dev.js",
     "test:coverage": "jest --coverage",
-    "cypress:cli": "npx cypress run-ct",
-    "cypress:ui": "npx cypress open-ct"
+    "cypress:cli": "node wrapCypress.ts cypress  run --component",
```
Comment (Thebarda) [inline package.json]:
```suggestion
    "cypress:cli": "cypress run --component --browser=chrome",
```
We don't need to use the wrapCypress.ts

**Review by zguennoune02 — DISMISSED** [review]

**Review by Thebarda — APPROVED** [review]

**Review by bdauria — APPROVED** [review]

**Review by zguennoune02 — APPROVED** [review]

**Pull request merged**